compiler/rustc_typeck/src/check/method/confirm.rs

```diff
@@ -182,10 +182,15 @@ impl<'a, 'tcx> ConfirmContext<'a, 'tcx> {
                     target,
                 });
 
-                if let Some(unsize_target) = unsize {
+                if *unsize {
```

Comment:
can you change l168 from `match &pick.autoref_or_ptr_adjustment {` to `match pick.autoref_or_ptr_adjustment {`?